src/frontend/src/optimizer/plan_node/logical_join.rs

```diff
@@ -502,6 +502,10 @@ impl LogicalJoin {
         let order_col_ids = table_desc.order_column_ids();
         let order_key = table_desc.order_column_indices();
         let dist_key = table_desc.distribution_key.clone();
+        // Avoid to lookup a table with a singleton distribution.
```

Comment:
May I ask why we ban this? Sounds feasible technically. 🤔

Reply:
You are right. The distributed lookup join has already supported it, but the local lookup join failed.